### DISCUSSION TITLE
Playing playlists

### DISCUSSION THREAD

**Discussion by xaeiougit** [pystardust/ytfzf#308] [Ideas]
I was trying to get ytfzf to play playlists.   Some playlists are large, and youtube doesn't provide an elegant way to search within them. 

I couldn't find any combination of ytfzf options to import a playlist so you could then search within it.  

The biggest complication I found is AFAIK the json from youtube for playlists won't include view_counts, date, and description.  One approach is to grab those dynamically as you navigate the search results and cache the data to disk (eg with a per-video json file)  But this makes the interface a bit clunky until the cache is built.  Another way is to prebuild a cache for a playlist before you pass it to ytfzl.   I used youtube-dl as a quick and dirty way to build the cache and then modded thumbnail_video_info_text in my conf.sh to use it.   For example:

` "youtube-dl --skip-download --print-json --no-warnings "https://www.youtube.com/watch?v=$shorturl" > "$jfile"`

Alternatively you could just ignore these data items for playlists altogether.  It's nice to see viewcounts at least tho.

Anyway the mods seem to work ok.   I haven't tested them much yet but I find it useful and am happy to share what I have :)

EDIT:  I've noticed my current approach suffers from the "infinite scrolling" problem for large playlists containing > 100 items...I am investigating to see if there's a reasonably efficient way around this.

ytfzf -t -X "https://www.youtube.com/playlist?list=PLGBuKfnErZlAkaUUy57-mR97f8SBgMNHh"
![playlist](https://user-images.githubusercontent.com/10321345/123645546-a27cc000-d850-11eb-84e2-6bbfe7264862.png)

**Comment by Euro20179** [comment]
On ytfzf versions 2.0 and above, you can do `ytfzf --type=playlist <playlist-search>` and when you select one it'll open a new menu with all the playlist items.

**Comment by xaeiougit** [comment]
Wow thanks, looks like lots of nice improvements in this new version!  I'll download the latest now and have a play, I'd love to be able to retire my modded version and move back onto the standard release.

**Comment by Euro20179** [comment]
> Wow thanks, looks like lots of nice improvements in this new version! I'll download the latest now and have a play, I'd love to be able to retire my modded version and move back onto the standard release.

Glad you like it, feel free to leave any feedback on what you think is missing, or is broken.
